Flaky Test: TestLocalOrchestration

```
[+] Shutdown all nodes
level=info ts="Thu, 15 Apr 2021 19:16:21 UTC" call=drand.go:292 beacon=stop
level=debug ts="Thu, 15 Apr 2021 19:16:21 UTC" call=:0 beacon_loop=finished
ERROR: 2021/04/15 19:16:21 [core] grpc: server failed to encode response:  rpc error: code = Internal desc = grpc: error while marshaling: proto: Marshal called with nil
level=error ts="Thu, 15 Apr 2021 19:16:21 UTC" call=panic.go:971 drand="failed to shutdown" err="rpc error: code = Unavailable desc = transport is closing"
level=debug ts="Thu, 15 Apr 2021 19:16:21 UTC" call=:0 beacon_loop=finished
level=info ts="Thu, 15 Apr 2021 19:16:21 UTC" call=drand.go:292 beacon=stop
level=info ts="Thu, 15 Apr 2021 19:16:21 UTC" call=:0 aggregated_beacon=3
level=info ts="Thu, 15 Apr 2021 19:16:21 UTC" call=chain.go:180 NEW_BEACON_STORED="{ round: 3, sig: a41656, prevSig: 897a79 }" time_discrepancy_ms=1234.217252
level=debug ts="Thu, 15 Apr 2021 19:16:21 UTC" call=:0 ignoring_partial=3 last_beacon_stored=3
ERROR: 2021/04/15 19:16:21 [core] grpc: server failed to encode response:  rpc error: code = Internal desc = grpc: error while marshaling: proto: Marshal called with nil
level=error ts="Thu, 15 Apr 2021 19:16:21 UTC" call=panic.go:971 drand="failed to shutdown" err="rpc error: code = Unavailable desc = transport is closing"
level=info ts="Thu, 15 Apr 2021 19:16:21 UTC" call=drand.go:292 beacon=stop
level=debug ts="Thu, 15 Apr 2021 19:16:21 UTC" call=:0 beacon_loop=finished
ERROR: 2021/04/15 19:16:21 [core] grpc: server failed to encode response:  rpc error: code = Internal desc = grpc: error while marshaling: proto: Marshal called with nil
level=error ts="Thu, 15 Apr 2021 19:16:21 UTC" call=panic.go:971 drand="failed to shutdown" err="rpc error: code = Unavailable desc = transport is closing"
panic: [-] Inconsistent beacon rounds between nodes
/usr/local/go/src/testing/testing.go:1143 +0x49f
/usr/local/go/src/testing/testing.go:1146 +0x695
/usr/local/go/src/runtime/panic.go:971 +0x499
/go/src/github.com/drand/drand/demo/lib/orchestrator.go:284 +0x1b65
/go/src/github.com/drand/drand/demo/lib/orchestrator.go:239 +0x145
/go/src/github.com/drand/drand/demo/demo_test.go:16 +0x158
/usr/local/go/src/testing/testing.go:1193 +0x203
/usr/local/go/src/testing/testing.go:1238 +0x5d8
```

Some extra logs:

```
 === RUN   TestLocalOrchestration
[+] Simulation global folder: /tmp/drand-full
2022/03/01 14:05:06 written cert.pem
2022/03/01 14:05:06 written key.pem
	- Created node 127.0.0.1:34241 at /tmp/drand-full
2022/03/01 14:05:07 written cert.pem
2022/03/01 14:05:07 written key.pem
	- Created node 127.0.0.1:43353 at /tmp/drand-full
2022/03/01 14:05:08 written cert.pem
2022/03/01 14:05:08 written key.pem
	- Created node 127.0.0.1:46673 at /tmp/drand-full
[+] Starting all nodes
	- Starting node 127.0.0.1:34241
Saved the key : 127.0.0.1:34241 at /tmp/drand-full/node-1/key/drand_id.public
level=info ts="Tue, 01 Mar 2022 14:05:08 UTC" call=drand.go:69 network=init insecure=false
level=info ts="Tue, 01 Mar 2022 14:05:08 UTC" call=drand.go:69 private_listen=127.0.0.1:34241 control_port=127.0.0.1:44129 public_listen=127.0.0.1:36471 folder=/tmp/drand-full/node-1
	- Starting node 127.0.0.1:46673
Saved the key : 127.0.0.1:46673 at /tmp/drand-full/node-3/key/drand_id.public
level=info ts="Tue, 01 Mar 2022 14:05:08 UTC" call=drand.go:69 network=init insecure=false
level=warn ts="Tue, 01 Mar 2022 14:05:08 UTC" call=gateway.go:59 grpcListener="failed metrics registration" err="duplicate metrics collector registration attempted"
level=info ts="Tue, 01 Mar 2022 14:05:08 UTC" call=drand.go:69 private_listen=127.0.0.1:46673 control_port=127.0.0.1:36419 public_listen=127.0.0.1:39175 folder=/tmp/drand-full/node-3
	- Starting node 127.0.0.1:43353
Saved the key : 127.0.0.1:43353 at /tmp/drand-full/node-2/key/drand_id.public
level=info ts="Tue, 01 Mar 2022 14:05:08 UTC" call=drand.go:69 network=init insecure=false
level=warn ts="Tue, 01 Mar 2022 14:05:08 UTC" call=gateway.go:59 grpcListener="failed metrics registration" err="duplicate metrics collector registration attempted"
level=info ts="Tue, 01 Mar 2022 14:05:08 UTC" call=drand.go:69 private_listen=127.0.0.1:43353 control_port=127.0.0.1:42373 public_listen=127.0.0.1:40521 folder=/tmp/drand-full/node-2
[+] Running DKG for all nodes
	- Running DKG for leader node 127.0.0.1:34241
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=server.go:1297 init_dkg=begin time=1646143509 leader=true
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=control_grpc.pb.go:276 reshare=newmanager
	- Running DKG for node 127.0.0.1:43353
	- Running DKG for node 127.0.0.1:46673
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=control_grpc.pb.go:276 init_dkg=begin leader=false
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=control_grpc.pb.go:276 init_dkg=begin leader=false
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=control_grpc.pb.go:276 init_dkg=send_key leader=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=control_grpc.pb.go:276 init_dkg=send_key leader=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:211 setup=received_new_key id="{127.0.0.1:43353 - bls12-381.G1: a092ccbaa6bda5c75fac2ddfa50ee0240f2a8f16dd702f5ce73903f45e8d509a1825ddb4ac458431c93f7327b6614042}"
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=:0 setup=added key="{127.0.0.1:43353 - bls12-381.G1: a092ccbaa6bda5c75fac2ddfa50ee0240f2a8f16dd702f5ce73903f45e8d509a1825ddb4ac458431c93f7327b6614042}" have=2/3
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=control_grpc.pb.go:276 init_dkg=wait_group
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:211 setup=received_new_key id="{127.0.0.1:46673 - bls12-381.G1: 866f7a07de5170c2d00be8a90439e44d00b17e9fead86cf8f746fabe965d297c698dc45e7e384a753c1b1f5d9d908820}"
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=:0 setup=added key="{127.0.0.1:46673 - bls12-381.G1: 866f7a07de5170c2d00be8a90439e44d00b17e9fead86cf8f746fabe965d297c698dc45e7e384a753c1b1f5d9d908820}" have=3/3
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 setup=created_group
Generated group:
Threshold = 2
Period = "4s"
CatchupPeriod = "0s"
GenesisTime = 1646143512
TransitionTime = 0
GenesisSeed = "6572fd5b35ee12b6286e22177817a3460597211a0c1b5777176ad8ce279ae390"

[[Nodes]]
  Address = "127.0.0.1:46673"
  Key = "866f7a07de5170c2d00be8a90439e44d00b17e9fead86cf8f746fabe965d297c698dc45e7e384a753c1b1f5d9d908820"
  TLS = true
  Signature = "945f044ac3581e1112349ca6180d2ea3a7e3fbff93caf04ac09a7a56c28ff92e272240732cb68872987aecb31032ffaf0f1570c85f3ca50c46e7253873dc56926cd4c46d0b58aab458f68fb318d9ad000422c751b6d58e1eaaa4e2828c7f299b"
  Index = 0

[[Nodes]]
  Address = "127.0.0.1:43353"
  Key = "a092ccbaa6bda5c75fac2ddfa50ee0240f2a8f16dd702f5ce73903f45e8d509a1825ddb4ac458431c93f7327b6614042"
  TLS = true
  Signature = "89bf3e3362da3e8d1afd5cfb7e8fb76cd1db9a6d9ba6af1c09f831de8ff3065e6122747554ef6077fa947d4073888f7f0eef6091b416bebb4fcd285f419b5530f5e6e9352da4996e70c0b74bba5e8c9278b96394bf04b0cc46d0c202ceb08b7a"
  Index = 1

[[Nodes]]
  Address = "127.0.0.1:34241"
  Key = "a2a5c897f5eb0eae74edcb802205c6791a4e2de62ff0d9efde38c61985e498f9171791cf18a978ad7324d9a0e3d0a4ff"
  TLS = true
  Signature = "a05680debdefbd6535e66adde2e4e14ddfa715c1b3c34793367764e7ba13b934023b878afcc5b861d9be8af351d2ea1b0b75b67f10d46aa040413dd33f4f4db3079009eee08c12d3eefc33abcf2c365da29b25b59efc35826b411cb8ba3cc5d4"
  Index = 2

level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=control_grpc.pb.go:276 init_dkg=setup_phase keys_received=[127.0.0.1:46673-127.0.0.1:43353-127.0.0.1:34241]
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=control_grpc.pb.go:276 init_dkg=wait_group
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=server_metrics.go:107 push_group=received_new
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=server_metrics.go:107 push_group=received_new
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=drand_control.go:44 init_dkg=received_group
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=drand_control.go:44 init_dkg=wait_dkg_end
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=drand_control.go:365 waiting_dkg_end=2022-03-01T14:05:09.767846327Z
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=control_grpc.pb.go:276 push_dkg=sending_group success_to=127.0.0.1:43353 left=1
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=drand_control.go:44 init_dkg=received_group
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=control_grpc.pb.go:276 push_dkg=sending_group success_to=127.0.0.1:46673 left=0
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=control_grpc.pb.go:276 push_dkg=sending_group status="all succeeded"
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=drand_control.go:44 init_dkg=wait_dkg_end
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=drand_control.go:365 waiting_dkg_end=2022-03-01T14:05:09.771893059Z
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=control_grpc.pb.go:276 init_dkg=START_DKG
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=control_grpc.pb.go:276 init_dkg=wait_dkg_end
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=drand_control.go:66 waiting_dkg_end=2022-03-01T14:05:09.773345243Z
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:205
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:288
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:206 echoBroadcast=push deal=6fa238c255
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=server_metrics.go:107 init_dkg="START DKG" signalfromleader=127.0.0.1:45020 group=6572fd5b35ee12b6286e22177817a3460597211a0c1b5777176ad8ce279ae390
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:205
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="received new packet to echoBroadcast" from=127.0.0.1:45020 type=*dkg.DealBundle
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:288
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:206 echoBroadcast=push deal=4530c291e2
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:45784 type=*dkg.DealBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="received new packet to echoBroadcast" from=127.0.0.1:45784 type=*dkg.DealBundle
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=server_metrics.go:107 init_dkg="START DKG" signalfromleader=127.0.0.1:59894 group=6572fd5b35ee12b6286e22177817a3460597211a0c1b5777176ad8ce279ae390
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="received new packet to echoBroadcast" from=127.0.0.1:59894 type=*dkg.DealBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:34241
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:205
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:45020 type=*dkg.DealBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="received new packet to echoBroadcast" from=127.0.0.1:59894 type=*dkg.DealBundle
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:288
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:206 echoBroadcast=push deal=e23213e4a7
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:231
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:43353
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:295
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:304
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:182 echoBroadcast=push response="ShareHolder 1: [{dealer 0, status true},{dealer 1, status true},{dealer 2, status true}]"
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:45782 type=*dkg.DealBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:59910 type=*dkg.DealBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="received new packet to echoBroadcast" from=127.0.0.1:45782 type=*dkg.ResponseBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:59910 type=*dkg.DealBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:59894 type=*dkg.ResponseBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:45782 type=*dkg.DealBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="received new packet to echoBroadcast" from=127.0.0.1:45020 type=*dkg.ResponseBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:59910 type=*dkg.ResponseBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:45784 type=*dkg.ResponseBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:45040 type=*dkg.ResponseBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="received new packet to echoBroadcast" from=127.0.0.1:45040 type=*dkg.DealBundle
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:231
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:45040 type=*dkg.DealBundle
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:295
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:304
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:182 echoBroadcast=push response="ShareHolder 0: [{dealer 0, status true},{dealer 1, status true},{dealer 2, status true}]"
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="received new packet to echoBroadcast" from=127.0.0.1:45784 type=*dkg.DealBundle
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:231
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:59910 type=*dkg.DealBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:45782 type=*dkg.DealBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:45020 type=*dkg.DealBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:45040 type=*dkg.DealBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="received new packet to echoBroadcast" from=127.0.0.1:45784 type=*dkg.ResponseBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:59894 type=*dkg.DealBundle
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:295
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:304
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:182 echoBroadcast=push response="ShareHolder 2: [{dealer 0, status true},{dealer 1, status true},{dealer 2, status true}]"
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:243
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:311
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:43353
crypto store: saving private share in /tmp/drand-full/node-1/groups/dist_key.private
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=drand_control.go:66 dkg_end=2022-03-01T14:05:09.882292897Z certified=3 list="[{addr: 127.0.0.1:46673, idx: 0, pub: bls12-381.G1: 866f7a07de5170c2d00be8a90439e44d00b17e9fead86cf8f746fabe965d297c698dc45e7e384a753c1b1f5d9d908820},{addr: 127.0.0.1:43353, idx: 1, pub: bls12-381.G1: a092ccbaa6bda5c75fac2ddfa50ee0240f2a8f16dd702f5ce73903f45e8d509a1825ddb4ac458431c93f7327b6614042},{addr: 127.0.0.1:34241, idx: 2, pub: bls12-381.G1: a2a5c897f5eb0eae74edcb802205c6791a4e2de62ff0d9efde38c61985e498f9171791cf18a978ad7324d9a0e3d0a4ff}]"
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=control_grpc.pb.go:276 init_dkg=dkg_done starting_beacon_time=1646143512 now=1646143509
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:45020 type=*dkg.ResponseBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="received new packet to echoBroadcast" from=127.0.0.1:59910 type=*dkg.ResponseBundle
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=:0 beacon_start=2022-03-01T14:05:09.889049885Z catchup=false
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 beacon=start
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=:0 run_round=wait until=1646143512
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:45040 type=*dkg.ResponseBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" errorto=127.0.0.1:34241 err:="rpc error: code = Unknown desc = drand: no dkg running"
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="ignoring duplicate packet" from=127.0.0.1:59894 type=*dkg.ResponseBundle
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="received new packet to echoBroadcast" from=127.0.0.1:59894 type=*dkg.ResponseBundle
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:243
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:311
crypto store: saving private share in /tmp/drand-full/node-2/groups/dist_key.private
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" errorto=127.0.0.1:34241 err:="rpc error: code = Unknown desc = drand: no dkg running"
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=drand_control.go:365 dkg_end=2022-03-01T14:05:09.902231641Z certified=3 list="[{addr: 127.0.0.1:46673, idx: 0, pub: bls12-381.G1: 866f7a07de5170c2d00be8a90439e44d00b17e9fead86cf8f746fabe965d297c698dc45e7e384a753c1b1f5d9d908820},{addr: 127.0.0.1:43353, idx: 1, pub: bls12-381.G1: a092ccbaa6bda5c75fac2ddfa50ee0240f2a8f16dd702f5ce73903f45e8d509a1825ddb4ac458431c93f7327b6614042},{addr: 127.0.0.1:34241, idx: 2, pub: bls12-381.G1: a2a5c897f5eb0eae74edcb802205c6791a4e2de62ff0d9efde38c61985e498f9171791cf18a978ad7324d9a0e3d0a4ff}]"
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=drand_control.go:44 init_dkg=dkg_done starting_beacon_time=1646143512 now=1646143509
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol_grpc.pb.go:247 echoBroadcast="received new packet to echoBroadcast" from=127.0.0.1:45020 type=*dkg.ResponseBundle
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:243
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=protocol.go:311
crypto store: saving private share in /tmp/drand-full/node-3/groups/dist_key.private
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=drand_control.go:365 dkg_end=2022-03-01T14:05:09.906603692Z certified=3 list="[{addr: 127.0.0.1:46673, idx: 0, pub: bls12-381.G1: 866f7a07de5170c2d00be8a90439e44d00b17e9fead86cf8f746fabe965d297c698dc45e7e384a753c1b1f5d9d908820},{addr: 127.0.0.1:43353, idx: 1, pub: bls12-381.G1: a092ccbaa6bda5c75fac2ddfa50ee0240f2a8f16dd702f5ce73903f45e8d509a1825ddb4ac458431c93f7327b6614042},{addr: 127.0.0.1:34241, idx: 2, pub: bls12-381.G1: a2a5c897f5eb0eae74edcb802205c6791a4e2de62ff0d9efde38c61985e498f9171791cf18a978ad7324d9a0e3d0a4ff}]"
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=drand_control.go:44 init_dkg=dkg_done starting_beacon_time=1646143512 now=1646143509
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=:0 beacon_start=2022-03-01T14:05:09.912225815Z catchup=false
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 beacon=start
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=:0 beacon_start=2022-03-01T14:05:09.912715243Z catchup=false
level=info ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 beacon=start
	 FINISHED DKG
	 FINISHED DKG
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=:0 run_round=wait until=1646143512
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=:0 run_round=wait until=1646143512
[+] Nodes finished running DKG. Checking keys...
[+] Checking if chain info is present on all nodes...
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" to=127.0.0.1:46673
	- Node 127.0.0.1:34241 has chain-info 83e4
	- Node 127.0.0.1:43353 has chain-info 83e4
	- Node 127.0.0.1:46673 has chain-info 83e4
[+] Chain info are present on all nodes. DKG finished.
[+] Checking all created group file with collective key
	- Overwrite group with distributed key to  /tmp/drand-full/group.toml
[+] Sleeping 2 until genesis happens
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" errorto=127.0.0.1:34241 err:="rpc error: code = Unknown desc = drand: no dkg running"
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" errorto=127.0.0.1:43353 err:="rpc error: code = Unknown desc = drand: no dkg running"
level=debug ts="Tue, 01 Mar 2022 14:05:09 UTC" call=asm_amd64.s:1581 echoBroadcast="sending out" errorto=127.0.0.1:46673 err:="rpc error: code = Unknown desc = drand: no dkg running"
[+] Sleeping 3s after genesis - leaving some time for rounds 
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 beacon_loop=new_round round=1 lastbeacon=0
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=asm_amd64.s:1581 broadcast_partial=1 from_prev_sig=6572fd msg_sign=b57fa5
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 store_partial=127.0.0.1:34241 round=1 len_partials=1/2
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 beacon_round=1 send_to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 beacon_round=1 send_to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:59894 round=1
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:45020 round=1
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:45020 index=2 prev_sig=6572fd curr_round=1 msg_sign=b57fa5 from_node=127.0.0.1:34241 status=OK
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 store_partial=127.0.0.1:45020 round=1 len_partials=1/2
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 beacon_loop=new_round round=1 lastbeacon=0
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:59894 index=2 prev_sig=6572fd curr_round=1 msg_sign=b57fa5 from_node=127.0.0.1:34241 status=OK
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 beacon_loop=new_round round=1 lastbeacon=0
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=asm_amd64.s:1581 broadcast_partial=1 from_prev_sig=6572fd msg_sign=b57fa5
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 beacon_round=1 send_to=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 store_partial=127.0.0.1:59894 round=1 len_partials=1/2
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 store_partial=127.0.0.1:43353 round=1 len_partials=2/2
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=asm_amd64.s:1581 broadcast_partial=1 from_prev_sig=6572fd msg_sign=b57fa5
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 beacon_round=1 send_to=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 store_partial=127.0.0.1:46673 round=1 len_partials=2/2
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 beacon_round=1 send_to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 beacon_round=1 send_to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:45040 round=1
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:59910 round=1
level=info ts="Tue, 01 Mar 2022 14:05:12 UTC" call=:0 aggregated_beacon=1
level=debug ts="Tue, 01 Mar 2022 14:05:12 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:45784 round=1
level=debug ts="Tue, 01 Mar 2022 14:05:13 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:45782 round=1
level=debug ts="Tue, 01 Mar 2022 14:05:13 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:45040 index=1 prev_sig=6572fd curr_round=1 msg_sign=b57fa5 from_node=127.0.0.1:43353 status=OK
level=debug ts="Tue, 01 Mar 2022 14:05:13 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:59910 index=0 prev_sig=6572fd curr_round=1 msg_sign=b57fa5 from_node=127.0.0.1:46673 status=OK
level=debug ts="Tue, 01 Mar 2022 14:05:13 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:45784 index=0 prev_sig=6572fd curr_round=1 msg_sign=b57fa5 from_node=127.0.0.1:46673 status=OK
level=debug ts="Tue, 01 Mar 2022 14:05:13 UTC" call=:0 store_partial=127.0.0.1:45784 round=1 len_partials=2/2
level=debug ts="Tue, 01 Mar 2022 14:05:13 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:45782 index=1 prev_sig=6572fd curr_round=1 msg_sign=b57fa5 from_node=127.0.0.1:43353 status=OK
level=info ts="Tue, 01 Mar 2022 14:05:13 UTC" call=:0 aggregated_beacon=1
level=info ts="Tue, 01 Mar 2022 14:05:13 UTC" call=chain.go:180 NEW_BEACON_STORED="{ round: 1, sig: a40896, prevSig: 6572fd }" time_discrepancy_ms=1078.743491
level=debug ts="Tue, 01 Mar 2022 14:05:13 UTC" call=:0 ignoring_partial=1 last_beacon_stored=1
level=debug ts="Tue, 01 Mar 2022 14:05:13 UTC" call=:0 beacon_loop=catchupmode last_is=1 current=1 catchup_launch=false
level=info ts="Tue, 01 Mar 2022 14:05:13 UTC" call=chain.go:180 NEW_BEACON_STORED="{ round: 1, sig: a40896, prevSig: 6572fd }" time_discrepancy_ms=1083.676374
level=debug ts="Tue, 01 Mar 2022 14:05:13 UTC" call=:0 ignoring_partial=1 last_beacon_stored=1
level=debug ts="Tue, 01 Mar 2022 14:05:13 UTC" call=:0 beacon_loop=catchupmode last_is=1 current=1 catchup_launch=false
level=info ts="Tue, 01 Mar 2022 14:05:13 UTC" call=:0 aggregated_beacon=1
level=info ts="Tue, 01 Mar 2022 14:05:13 UTC" call=chain.go:180 NEW_BEACON_STORED="{ round: 1, sig: a40896, prevSig: 6572fd }" time_discrepancy_ms=1150.859331
level=debug ts="Tue, 01 Mar 2022 14:05:13 UTC" call=:0 ignoring_partial=1 last_beacon_stored=1
level=debug ts="Tue, 01 Mar 2022 14:05:13 UTC" call=:0 beacon_loop=catchupmode last_is=1 current=1 catchup_launch=false
[+] Sleeping 5s to reach round 2 + 3s
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 beacon_loop=new_round round=2 lastbeacon=1
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=asm_amd64.s:1581 broadcast_partial=2 from_prev_sig=a40896 msg_sign=65f479
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 beacon_round=2 send_to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 store_partial=127.0.0.1:34241 round=2 len_partials=1/2
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 beacon_round=2 send_to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:45020 round=2
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:59894 round=2
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:45020 index=2 prev_sig=a40896 curr_round=2 msg_sign=65f479 from_node=127.0.0.1:34241 status=OK
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:59894 index=2 prev_sig=a40896 curr_round=2 msg_sign=65f479 from_node=127.0.0.1:34241 status=OK
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 store_partial=127.0.0.1:45020 round=2 len_partials=1/2
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 store_partial=127.0.0.1:59894 round=2 len_partials=1/2
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 beacon_loop=new_round round=2 lastbeacon=1
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 beacon_loop=new_round round=2 lastbeacon=1
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=asm_amd64.s:1581 broadcast_partial=2 from_prev_sig=a40896 msg_sign=65f479
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 store_partial=127.0.0.1:43353 round=2 len_partials=2/2
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=asm_amd64.s:1581 broadcast_partial=2 from_prev_sig=a40896 msg_sign=65f479
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 beacon_round=2 send_to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 store_partial=127.0.0.1:46673 round=2 len_partials=2/2
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 beacon_round=2 send_to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 beacon_round=2 send_to=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 beacon_round=2 send_to=127.0.0.1:34241
level=info ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 aggregated_beacon=2
level=info ts="Tue, 01 Mar 2022 14:05:16 UTC" call=chain.go:180 NEW_BEACON_STORED="{ round: 2, sig: 95e2b3, prevSig: a40896 }" time_discrepancy_ms=959.97551
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 beacon_loop=catchupmode last_is=2 current=2 catchup_launch=false
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:59910 round=2
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:45782 round=2
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:59910 index=0 prev_sig=a40896 curr_round=2 msg_sign=65f479 from_node=127.0.0.1:46673 status=OK
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:45782 index=1 prev_sig=a40896 curr_round=2 msg_sign=65f479 from_node=127.0.0.1:43353 status=OK
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 store_partial=127.0.0.1:45782 round=2 len_partials=2/2
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:45040 round=2
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:45040 index=1 prev_sig=a40896 curr_round=2 msg_sign=65f479 from_node=127.0.0.1:43353 status=OK
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 ignoring_partial=2 last_beacon_stored=2
level=info ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 aggregated_beacon=2
level=info ts="Tue, 01 Mar 2022 14:05:16 UTC" call=chain.go:180 NEW_BEACON_STORED="{ round: 2, sig: 95e2b3, prevSig: a40896 }" time_discrepancy_ms=986.846035
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 ignoring_partial=2 last_beacon_stored=2
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=:0 beacon_loop=catchupmode last_is=2 current=2 catchup_launch=false
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:45784 round=2
level=debug ts="Tue, 01 Mar 2022 14:05:16 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:45784 index=0 prev_sig=a40896 curr_round=2 msg_sign=65f479 from_node=127.0.0.1:46673 status=OK
level=info ts="Tue, 01 Mar 2022 14:05:17 UTC" call=:0 aggregated_beacon=2
level=info ts="Tue, 01 Mar 2022 14:05:17 UTC" call=chain.go:180 NEW_BEACON_STORED="{ round: 2, sig: 95e2b3, prevSig: a40896 }" time_discrepancy_ms=1011.662543
level=debug ts="Tue, 01 Mar 2022 14:05:17 UTC" call=:0 ignoring_partial=2 last_beacon_stored=2
level=debug ts="Tue, 01 Mar 2022 14:05:17 UTC" call=:0 beacon_loop=catchupmode last_is=2 current=2 catchup_launch=false
level=debug ts="Tue, 01 Mar 2022 14:05:19 UTC" call=asm_amd64.s:1581 phaser_finished=deal
level=debug ts="Tue, 01 Mar 2022 14:05:19 UTC" call=asm_amd64.s:1581 phaser_finished=deal
level=debug ts="Tue, 01 Mar 2022 14:05:19 UTC" call=asm_amd64.s:1581 phaser_finished=deal
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=:0 beacon_loop=new_round round=3 lastbeacon=2
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=asm_amd64.s:1581 broadcast_partial=3 from_prev_sig=95e2b3 msg_sign=2d58a8
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=:0 beacon_round=3 send_to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=:0 beacon_round=3 send_to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=:0 store_partial=127.0.0.1:34241 round=3 len_partials=1/2
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:59894 round=3
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:45020 round=3
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:45020 index=2 prev_sig=95e2b3 curr_round=3 msg_sign=2d58a8 from_node=127.0.0.1:34241 status=OK
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=:0 beacon_loop=new_round round=3 lastbeacon=2
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=asm_amd64.s:1581 broadcast_partial=3 from_prev_sig=95e2b3 msg_sign=2d58a8
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:59894 index=2 prev_sig=95e2b3 curr_round=3 msg_sign=2d58a8 from_node=127.0.0.1:34241 status=OK
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=:0 beacon_loop=new_round round=3 lastbeacon=2
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=asm_amd64.s:1581 broadcast_partial=3 from_prev_sig=95e2b3 msg_sign=2d58a8
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=:0 store_partial=127.0.0.1:59894 round=3 len_partials=1/2
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=:0 store_partial=127.0.0.1:43353 round=3 len_partials=2/2
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=:0 store_partial=127.0.0.1:45020 round=3 len_partials=1/2
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=:0 store_partial=127.0.0.1:46673 round=3 len_partials=2/2
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=:0 beacon_round=3 send_to=127.0.0.1:43353
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=:0 beacon_round=3 send_to=127.0.0.1:46673
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=:0 beacon_round=3 send_to=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=:0 beacon_round=3 send_to=127.0.0.1:34241
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:45784 round=3
level=debug ts="Tue, 01 Mar 2022 14:05:20 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:45782 round=3
[+] Checking randomness beacon for round 3 via CLI
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:45040 round=3
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:45784 index=0 prev_sig=95e2b3 curr_round=3 msg_sign=2d58a8 from_node=127.0.0.1:46673 status=OK
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=:0 store_partial=127.0.0.1:45784 round=3 len_partials=2/2
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=protocol_grpc.pb.go:265 received=request from=127.0.0.1:59910 round=3
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:45782 index=1 prev_sig=95e2b3 curr_round=3 msg_sign=2d58a8 from_node=127.0.0.1:43353 status=OK
level=info ts="Tue, 01 Mar 2022 14:05:21 UTC" call=:0 aggregated_beacon=3
level=info ts="Tue, 01 Mar 2022 14:05:21 UTC" call=chain.go:180 NEW_BEACON_STORED="{ round: 3, sig: 835ab1, prevSig: 95e2b3 }" time_discrepancy_ms=1063.170886
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:59910 index=0 prev_sig=95e2b3 curr_round=3 msg_sign=2d58a8 from_node=127.0.0.1:46673 status=OK
level=info ts="Tue, 01 Mar 2022 14:05:21 UTC" call=:0 aggregated_beacon=3
level=info ts="Tue, 01 Mar 2022 14:05:21 UTC" call=chain.go:180 NEW_BEACON_STORED="{ round: 3, sig: 835ab1, prevSig: 95e2b3 }" time_discrepancy_ms=1069.215229
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=:0 ignoring_partial=3 last_beacon_stored=3
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=:0 beacon_loop=catchupmode last_is=3 current=3 catchup_launch=false
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=protocol_grpc.pb.go:265 process_partial_from=127.0.0.1:45040 index=1 prev_sig=95e2b3 curr_round=3 msg_sign=2d58a8 from_node=127.0.0.1:43353 status=OK
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=:0 ignoring_partial=3 last_beacon_stored=3
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=:0 beacon_loop=catchupmode last_is=3 current=3 catchup_launch=false
level=info ts="Tue, 01 Mar 2022 14:05:21 UTC" call=server_metrics.go:107 public_rand=127.0.0.1:60020 round=3 reply="{ round: 3, sig: 835ab1, prevSig: 95e2b3 }"
	 - Example command is: "unused"
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=server_metrics.go:107 public_rand=unstored_beacon round=3 from=127.0.0.1:45916
level=error ts="Tue, 01 Mar 2022 14:05:21 UTC" call=orchestrator.go:239 drand="can't get becon" err="rpc error: code = Unknown desc = can't retrieve beacon: beacon not found in database <nil>"
last index 2  vs current index  1
round:3  signature:"\x83Z\xb1\xb2\xbc\xf9,\xc3\xc4Q\\\xd3\xf5\x08\xef;7\xf1\xc7=\x81\x99#\x01x\xe8\x1d\x05r\xb7\xf9\xfb\x81)\xb4;\xc7Y\x15\x00-\x83\x83IS8\x02\xc1\t\x85\x86c\xe2\x9e\xfc;\xb5\xe7\x94\xd22\xbd\x04\xf2@F\xe2\xed\xe5{T\x16?\x9f\x00\xd3X\xb7\rb\x1f\x95,\x95\xbeŔ\xc7P\x9b\xe1v\xff\xa4\xff\xe8"  randomness:"\xd2\xe3f^\xe9\xbc\x1e\x98\xda'T\xf6\x8f\x14M\xd2F\xa2\x90\x01\xd0v\xad\xa5\xe0\x11k%r6\xbd\n"
<nil>
[+] Shutdown all nodes
	- Stop old node 127.0.0.1:34241
level=info ts="Tue, 01 Mar 2022 14:05:21 UTC" call=drand.go:292 beacon=stop
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=:0 beacon_loop=finished
ERROR: 2022/03/01 14:05:21 [core] grpc: server failed to encode response:  rpc error: code = Internal desc = grpc: error while marshaling: proto: Marshal called with nil
level=error ts="Tue, 01 Mar 2022 14:05:21 UTC" call=panic.go:1047 drand="failed to shutdown" err="rpc error: code = Unavailable desc = error reading from server: EOF"
	- Stop old node 127.0.0.1:43353
level=info ts="Tue, 01 Mar 2022 14:05:21 UTC" call=drand.go:292 beacon=stop
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=:0 beacon_loop=finished
ERROR: 2022/03/01 14:05:21 [core] grpc: server failed to encode response:  rpc error: code = Internal desc = grpc: error while marshaling: proto: Marshal called with nil
level=error ts="Tue, 01 Mar 2022 14:05:21 UTC" call=panic.go:1047 drand="failed to shutdown" err="rpc error: code = Unavailable desc = error reading from server: EOF"
	- Stop old node 127.0.0.1:46673
level=info ts="Tue, 01 Mar 2022 14:05:21 UTC" call=drand.go:292 beacon=stop
level=debug ts="Tue, 01 Mar 2022 14:05:21 UTC" call=:0 beacon_loop=finished
ERROR: 2022/03/01 14:05:21 [core] grpc: server failed to encode response:  rpc error: code = Internal desc = grpc: error while marshaling: proto: Marshal called with nil
level=error ts="Tue, 01 Mar 2022 14:05:21 UTC" call=panic.go:1047 drand="failed to shutdown" err="rpc error: code = Unavailable desc = error reading from server: EOF"
--- FAIL: TestLocalOrchestration (15.00s)
```

and at the end quite interesting:

```
panic: [-] Inconsistent beacon rounds between nodes [recovered]
	panic: [-] Inconsistent beacon rounds between nodes

goroutine 19 [running]:
testing.tRunner.func1.2({0xea7c40, 0x10d3470})
	/opt/hostedtoolcache/go/1.17.7/x64/src/testing/testing.go:1209 +0x36c
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.17.7/x64/src/testing/testing.go:1212 +0x3b6
panic({0xea7c40, 0x10d3470})
	/opt/hostedtoolcache/go/1.17.7/x64/src/runtime/panic.go:1047 +0x266
github.com/drand/drand/demo/lib.(*Orchestrator).checkBeaconNodes(0xc0002226e0, {0xc000484d80, 0x3, 0x0}, {0xc000039700, 0x1a}, 0x0)
	/home/runner/work/drand/drand/demo/lib/orchestrator.go:284 +0x1618
github.com/drand/drand/demo/lib.(*Orchestrator).CheckCurrentBeacon(0xc0002226e0, {0x0, 0x0, 0x0})
	/home/runner/work/drand/drand/demo/lib/orchestrator.go:239 +0xf0
github.com/drand/drand/demo.TestLocalOrchestration(0x0)
	/home/runner/work/drand/drand/demo/demo_test.go:16 +0x10c
testing.tRunner(0xc00021cb60, 0xfedf40)
	/opt/hostedtoolcache/go/1.17.7/x64/src/testing/testing.go:1259 +0x230
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.17.7/x64/src/testing/testing.go:1306 +0x727
FAIL	github.com/drand/drand/demo	15.033s
```

--- a/demo/lib/orchestrator.go
+++ b/demo/lib/orchestrator.go
@@ -292,14 +292,16 @@ func (e *Orchestrator) checkBeaconNodes(nodes []node.Node, group string, tryCurl
 				fmt.Printf("\t - Example command is: \"%s\"\n", cmd)
 				break
 			} else {
-				if !bytes.Equal(randResp.GetSignature(), rand.GetSignature()) {
-					panic("[-] Inconsistent beacon signature between nodes")
-				}
+				// we first check both are at the same round
 				if randResp.GetRound() != rand.GetRound() {
 					fmt.Println("[-] Mismatch between last index", lastIndex, " vs current index ", node.Index(), " - trying again in some time...")
 					time.Sleep(100 * time.Millisecond)
 					// we try again
 					continue
+				}
+				// then we check if the signatures match
+				if !bytes.Equal(randResp.GetSignature(), rand.GetSignature()) {
+					panic("[-] Inconsistent beacon signature between nodes")
 				}
 				// everything is good
 				break

